### PR TITLE
8326509: Clean up JNIEXPORT in Hotspot after JDK-8017234

### PR DESCRIPTION
**Pull request by magicus**
Once [JDK-8017234](https://bugs.openjdk.org/browse/JDK-8017234) has been integrated, it is possible to do some cleanup. The goal of [JDK-8017234](https://bugs.openjdk.org/browse/JDK-8017234) was to not change any behavior, even if that behavior seemed odd.

Now let's try to fix that. We can:

a) remove JNIEXPORT from c2v functions.
b) make debug.cpp functions exported similarly on all platforms.
c) remove JNIEXPORT from aarch64 asm debug function.

Note that this PR is [dependent on](https://mail.openjdk.org/pipermail/jdk-dev/2021-March/005232.html) https://github.com/openjdk/jdk/pull/17955.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8326509](https://bugs.openjdk.org/browse/JDK-8326509): Clean up JNIEXPORT in Hotspot after JDK-8017234 (**Bug** - P3)

### Reviewers
 * [Daniel Jeliński](https://openjdk.org/census#djelinski) (@djelinski - **Reviewer**)
 * [Julian Waters](https://openjdk.org/census#jwaters) (@TheShermanTanker - Committer) ⚠️ Review applies to [b6bd5f22](https://git.openjdk.org/jdk/pull/17967/files/b6bd5f223da19e8be066ba7d4a7063cfe8bc7a63)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17967/head:pull/17967` \
`$ git checkout pull/17967`

Update a local copy of the PR: \
`$ git checkout pull/17967` \
`$ git pull https://git.openjdk.org/jdk.git pull/17967/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17967`

View PR using the GUI difftool: \
`$ git pr show -t 17967`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17967.diff">https://git.openjdk.org/jdk/pull/17967.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17967#issuecomment-1959822984)

### CODE REVIEW

File: src/hotspot/cpu/aarch64/assembler_aarch64.cpp [hunk 1]
```diff
@@ -124,7 +119,7 @@ extern "C" {
       Disassembler::decode((address)start, (address)start + len);
   }
 
-  JNIEXPORT void das1(uintptr_t insn) {
+  void das1(uintptr_t insn) {
```
Comment (dholmes-ora) [inline src/hotspot/cpu/aarch64/assembler_aarch64.cpp:122]: Doesn't this need to be exported for debugging on Windows-Aarch64?

**Comment by dholmes-ora** [comment]
With the one exception above I agree with these choices going forward. I don't think it matters that exporting for the debugger is not actually necessary on some platforms, and so code simplicity wins out for me. That said the litmus test will be whether hermetic Java encounters any conflicts.

Thanks

File: src/hotspot/share/jvmci/vmStructs_jvmci.cpp [hunk 2]
```diff
@@ -1029,11 +1024,11 @@ int JVMCIVMStructs::localHotSpotVMAddresses_count() {
 }
 
 extern "C" {
-JNIEXPORT VMStructEntry* jvmciHotSpotVMStructs = JVMCIVMStructs::localHotSpotVMStructs;
-JNIEXPORT VMTypeEntry* jvmciHotSpotVMTypes = JVMCIVMStructs::localHotSpotVMTypes;
-JNIEXPORT VMIntConstantEntry* jvmciHotSpotVMIntConstants = JVMCIVMStructs::localHotSpotVMIntConstants;
-JNIEXPORT VMLongConstantEntry* jvmciHotSpotVMLongConstants = JVMCIVMStructs::localHotSpotVMLongConstants;
-JNIEXPORT VMAddressEntry* jvmciHotSpotVMAddresses = JVMCIVMStructs::localHotSpotVMAddresses;
+VMStructEntry* jvmciHotSpotVMStructs = JVMCIVMStructs::localHotSpotVMStructs;
```
Comment (djelinski) [inline src/hotspot/share/jvmci/vmStructs_jvmci.cpp:1027]: All these globals are unused now. You can remove them.

**Review by djelinski — COMMENTED** [review]
Nice cleanup!

File: src/hotspot/cpu/aarch64/assembler_aarch64.cpp (hunk 1 above)
Comment (magicus) [reply src/hotspot/cpu/aarch64/assembler_aarch64.cpp:122]: Ah, right, there is a Windows/aarch64 port. Forgot about it. :(

I don't know who maintains/cares about that port anymore. Pinging @mo-beck @lewurm @luhenry. Do anyone of you care about this anymore? If not, do you know if anyone has taken up the baton? (Or maybe it's time to retire the Windows/aarch64 port)

Comment (lewurm) [reply src/hotspot/cpu/aarch64/assembler_aarch64.cpp:122]: cc @karianna @gdams, is there a Windows/AArch64 owner?

> Or maybe it's time to retire the Windows/aarch64 port

I guess Windows/AArch64 as a platform hasn't really taken off _yet_, give it a few more years 😉 I certainly would not advocate for removing it.

Comment (lewurm) [reply src/hotspot/cpu/aarch64/assembler_aarch64.cpp:122]: Personally I have never used `das`/`das1` on any AArch64 port when debugging.  I guess it was somewhat handy when the backend was originally developed.  I would argue that the built-in disassembler of any debugger and `disnm` from `debug.cpp` are good enough, and thus `das`/`das1` can be removed.  What do you think @theRealAph ?

@

Comment (karianna) [reply src/hotspot/cpu/aarch64/assembler_aarch64.cpp:122]: Hi all - yes the Windows Arm port is maintained by Microsoft's JDK team. How can we help in this particular case?

Comment (karianna) [reply src/hotspot/cpu/aarch64/assembler_aarch64.cpp:122]: I assume you want to know if we use DAS/DAS1 for debugging purposes on that platform?

Comment (theRealAph) [reply src/hotspot/cpu/aarch64/assembler_aarch64.cpp:122]: I use `das` every day. In contrast, I don't know what `das1` is for.

Comment (magicus) [reply src/hotspot/cpu/aarch64/assembler_aarch64.cpp:122]: It does not sound like anyone object to the removal of `JNIEXPORT` for `das1`, then.

Or should I just remove the entire function, if it serves no purpose that any current maintainers know or care about?

Comment (luhenry) [reply src/hotspot/cpu/aarch64/assembler_aarch64.cpp:127]: You can probably remove the entire function given it's an indirection to `das` anyway. I remember using it for debugging during the first stages of the Windows-AArch64 port but I've long forgotten about it.

Comment (adinn) [reply src/hotspot/cpu/aarch64/assembler_aarch64.cpp:127]: @luhenry @magicus das1 was added when we were implementing the AArch64 port in order to help us integrate the debugger support we provided when running JITted code on our AArch64 simulator into gdb. Now that we have real hardware it is redundant.

Comment (magicus) [reply src/hotspot/cpu/aarch64/assembler_aarch64.cpp]: Thanks @adinn for the confirmation! I've now removed it.

**Comment by magicus** [comment]
There was a bit back and forth with this PR, but now I believe it is finally ready. Any reviewers?

File: src/hotspot/cpu/aarch64/assembler_aarch64.cpp (hunk 1 above)
Comment (brianjstafford) [reply src/hotspot/cpu/aarch64/assembler_aarch64.cpp]: Removal makes sense - thank you @lewurm, @luhenry, and @adinn for your input!

**Review by dholmes-ora — APPROVED** [review]
Seems fine to me. Thanks.

**Review by djelinski — APPROVED** [review]

**Pull request closed without merging**